robottelo/cleanup.py

```diff
@@ -17,3 +22,89 @@ def location_cleanup(loc_id=None):
 def org_cleanup(org_id=None):
     """Deletes the Org with the given id"""
     entities.Organization(id=org_id).delete()
+
+
+class EntitiesCleaner(object):
+    """Register and clean entities for cleanup using signals"""
+
+    def __init__(self, *types_to_cleanup):
+        self.cleanup_queue = defaultdict(deque)
+        self.deleted_entities = defaultdict(set)
+        self.types_to_cleanup = types_to_cleanup
+        self.logger = logging.getLogger('robottelo')
+        self.connect_cleanup_signals()
+
+    def connect_cleanup_signals(self):
+        """Connect signals to entity types specified for cleanup"""
+        for entity_type in self.types_to_cleanup:
+            self.logger.info('Registering %s signal for cleanup', entity_type)
+            signals.post_create.connect(self.register_entity_for_cleanup,
+                                        sender=entity_type)
+
+    def register_entity_for_cleanup(self, sender, entity, **kwargs):
+        """Put a new entity in the queue to be cleaned"""
+        self.logger.info(
+            'Adding {0}:{1} for cleanup_queue'.format(sender, entity.id))
+        self.cleanup_queue[entity.__class__.__name__].appendleft(entity)
+
+    def clean(self):
+        """This method is called in TearDownClass only when cleanup=true"""
+        default_org = entities.Organization(id=1)
```

Comment:
It would be nice to fetch the ID for the default organization instead of hard code it. I know the most of the time the ID will be 1 but still, you never know?